File: myBackend/app/models/accountModel.py
from pydantic import BaseModel
from enum import Enum


class AccountType(str, Enum):
    CASH = 'Cash'
    BANK = 'Bank'
    CREDIT_CARD = 'Credit Card'
    INVESTMENT = 'Investment'
    LOAN = 'Loan'


class Account(BaseModel):
    name: str
    balance: float = 0  # user can set balance if they want to
    currency_id: int
    account_type: AccountType

File: my_frontend/lib/models/account_model.dart
class Account {
  final int accountId;
  final String name;
  final double balance;
  final String accountType;
  final int currencyId;

  Account({
    required this.accountId,
    required this.name,
    required this.balance,
    required this.accountType,
    required this.currencyId,
  });

  factory Account.fromJson(Map<String, dynamic> json) => Account(
    accountId: json['account_id'],
    name: json['name'],
    balance: json['balance'].toDouble(),
    accountType: json['account_type'],
    currencyId: json['currency_id'],
  );
}